docs/api/v1.md

```diff
@@ -1,80 +1,263 @@
+
 
 ### `POST /debug/attach/bin/:path*`
+Attach to a binary, so it can be executed. Returns path to source code.
+
+
+Path parameters:
+ - `path` {string}: Path to binary to run and attach to.
+
+
+Responses:
+ - `200`: Successfully ran and attached to binary.
+     - Returns {[AttachInfo](#attachInfo)} 
+ - `400`: Invalid path.
+     - Returns {[Error](#error)}
+ - `404`: No binary exists at given path. 
+     - Returns {[Error](#error)}
+
 
 ### `GET /debug`
 Returns information about the attached process or executable.
 
+
+Responses:
+ - `200`: Is attached to process.
+     - Returns {[AttachInfo](#attachInfo)} 
+
+
+## Functions
+
+
 ### `GET /debug/functions`
-Returns a list of functions in the attached process.
+Returns a list of debuggable functions in the attached process.
+
+
+Responses:
+ - `200`: Successfully got list of functions.
+     - Returns {string[]}: List of function identifiers. 
+
 
 ### `GET /debug/functions/:function`
-Returns source code and input fields for this function.
+Returns information about the function, including source file path and input parameter types.
 
-### `PUT /debug/functions/:function/breakpoints`
-Sets a breakpoint on this function.
 
-### `DELETE /debug/functions/:function/breakpoints`
-Removes breakpoint on this function.
+Path parameters:
+ - `function` {string}: Function identifier.
+
+
+Responses:
+ - `200`: Got info for specified function.
+     - Returns {[Function](#function)} 
+ - `400`: Badly formatted function identifier.
+     - Returns {[Error](#error)}
+ - `404`: No function of that identifier found. 
+     - Returns {[Error](#error)}
+
+
+## Breakpoints
+
 
 ### `GET /debug/breakpoints`
 Lists breakpoints.
 
+
+Responses:
+ - `200`: Got list of breakpoints.
+     - Returns {[Breakpoint](#breakpoint)[]}: List of zero or more breakpoints set.
+
+
+### `PUT /debug/breakpoints/:function`
+Sets a breakpoint on this function.
+
+
+Path parameters:
+ - `function` {string}: Function identifier.
+
+
+Response:
+ - `200`: Successfully set breakpoint at function.
+     - Returns {[Breakpoint](#breakpoint)}
+ - `400`: Badly formatted function identifier.
+     - Returns {[Error](#error)}
+ - `404`: No function of that identifier found. 
+     - Returns {[Error](#error)}
+
+
+### `DELETE /debug/breakpoints/:function`
+Removes breakpoint on this function.
+
+
+Path parameters:
+ - `function` {string}: Function identifier.
+
+
+Response:
+ - `200`: Successfully removed breakpoint at function.
+ - `400`: Badly formatted function identifier.
+     - Returns {[Error](#error)}
+ - `404`: No function of that identifier found. 
+     - Returns {[Error](#error)}
+ - `410`: There was no breakpoint on that function to begin with.
+     - Returns {[Error](#error)}
+
+
+
+
+### `POST /debug/execute`
+Launches the process if it is not running or continues execution until the next breakpoint.
+Returns execution time, program output, and an [execution id](#execution) when the program encounters a breakpoint,
```

Comment:
I'll clarify the wording (also there's some inconsistencies with the formatting of this endpoint). The expected use case is that the when the user presses the "continue" button the we hit this endpoint, which returns immediately with an execution. The client uses this to call `/debug/execuction/:executionId/trace`, which will stream `stdout` of the program. The trace will end either when the process terminates, or execution hits a breakpoint. If it hits a breakpoint, the trace will be a `breakpoint` trace which contains `nextExecution`, so we can switch to "debug" view and immediately trace the next execution.